framework/web/UrlManager.php

```diff
@@ -428,6 +429,24 @@ public function createUrl($params)
     }
 
     /**
+     * Returns the value indicating whether result of [[createUrl()]] of rule should be cached in internal cache.
+     *
+     * @param UrlRuleInterface $rule
+     * @return bool|null `true` if result should be cached, `false` if not, `null` if rule does not provide info about
```

Comment:
Then this bug will still exists if custom rule does not provide info about `createStatus` - rule may return `false` if some params does not match, and then it will be skipped in all further URLs generation, even if different (matching) params are passed to `UrlManager::createUrl()`.